addons/Drop/drop.lua

```diff
@@ -0,0 +1,65 @@
+_addon.name = 'Drop'
+_addon.author = 'Arico'
+_addon.version = '1'
+_addon.commands = {'drop'}
+
+
+packets = require('packets')
+require('luau')
+res = require('resources')
+require('logger')
+
+check_inv = function(item_id)
+        for k, v in pairs(windower.ffxi.get_items().inventory) do
+            if type(v) == 'table' and v.id == item_id then
+                return true
+            end
+        end
+    return false
+end
+get_item_resource = function(item)
+    for k, v in pairs(res.items) do
+        if v.english:lower() == item:lower() or v.japanese:lower() == item:lower()  then
+            return v
+        end
+    end
+    return nil
+end
+
+drop_item = function(item_to_drop) 

```

Comment:
This function behaves almost the same as the `check_inv` function, only the loop body is different. I would skip the `check_inv` function entirely and call `drop_item` directly. This function can still return `true` or `false` based on if it actually dropped something. You can even combine that with `coroutine.sleep`, since that function also returns nothing, which evaluates to `false`. This makes the function design and invocation both easier.